src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,99 @@
-public class StringManipulation implements StringManipulationInterface {
+// Sean Michael CS410
+
+import java.util.Arrays;
+import java.util.StringTokenizer;
 
+public class StringManipulation implements StringManipulationInterface {
+    String string;
     @Override
     public String getString() {
         return null;
     }
 
     @Override
     public void setString(String string) {
+        this.string = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        StringTokenizer st = new StringTokenizer(string, " ");
+        return st.countTokens();
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        // if n < 0, throw illegal argument exception
+        if (n < 0) throw new IllegalArgumentException("n cannot be negative");
+        // if n > string.length(), throw out of bounds exception
+        else if (n > string.length()) throw new IndexOutOfBoundsException("n cannot exceed string length");
+        // if n == 0 the string is unchanged
+        else if (n == 0) return string;
+        // if every 1 char is being removed, all should be removed and either replaced with space or not
+        else if (n == 1) {
+            if (maintainSpacing) {
+                return " ".repeat(string.length());
+            }
+            else return "";
+        }
+        // otherwise, remove every nth character
+        else {
+            StringBuilder sb = new StringBuilder(string);
+            // start at end of string minus the remainder of the string divided by n, the "last" char to be removed
+            for (int i = string.length() - string.length() % n - 1; i >= 0; i -= n) {
+                if (maintainSpacing) {
+                    sb.replace(i,i+1," ");
+                }
+                else sb.deleteCharAt(i);
+            }
+            return sb.toString();
+        }
+
     }
 
     @Override
-    public String[] getSubStrings(int startWord, int endWord) {
-        return null;
```

Comment:
There is two getSubstrings methods here

Reply:
The - before the line indicates that it was removed from the last commit